app/src/main/java/org/simple/clinic/patient/PatientRepository.kt

```diff
@@ -61,9 +73,20 @@ class PatientRepository @Inject constructor(
           .toObservable()
     }
 
-    return database.patientSearchDao()
-        .search(nameToSearchableForm(query!!), dateOfBirthUpperBound, dateOfBirthLowerBound)
-        .toObservable()
+    val actualQuery = nameToSearchableForm(query!!)
+
+    return if (actualQuery.all { it.isDigit() } || includeFuzzyNameSearch.not()) {
+      database.patientSearchDao()
+          .search(actualQuery, dateOfBirthUpperBound, dateOfBirthLowerBound)
+          .toObservable()
+    } else {
+      database.patientSearchDao()
+          .search(actualQuery, dateOfBirthUpperBound, dateOfBirthLowerBound)
+          .zipWith(database.fuzzyPatientSearchDao().searchForPatientsWithNameLikeAndAgeWithin(actualQuery, dateOfBirthUpperBound, dateOfBirthLowerBound).toFlowable()) { searchResults, fuzzySearchResults ->
+            (fuzzySearchResults + searchResults).distinctBy { it.uuid }
+          }
+          .toObservable()
+    }
```

Comment:
Same question here. Not only is there too much happening at line [#85](https://github.com/simpledotorg/simple-android/blob/8b65e3b62e3e44d6ba437b4f13b20fa507da18e3/app/src/main/java/org/simple/clinic/patient/PatientRepository.kt#L85), but you have also gone past the margin.

What do you think about this?

```kotlin
val fuzzySearch = database.fuzzyPatientSearchDao()
    .searchForPatientsWithNameLikeAndAgeWithin(actualQuery, dateOfBirthUpperBound, dateOfBirthLowerBound)
    .toObservable()

database.patientSearchDao()
    .search(actualQuery, dateOfBirthUpperBound, dateOfBirthLowerBound)
    .toObservable()
    .zipWith(fuzzySearch)
    .map { (results, fuzzyResults) -> (fuzzyResults + results).distinctBy { it.uuid } }
```